app/dcr_monitor/tee_backend/confidential_space.go

```diff
@@ -135,9 +135,13 @@ func (c *TEEProviderGCPConfidentialSpace) getConfidentialSpaceInsertInstanceRequ
 	}
 
```

Comment:
please add some validation here (e.g., GCP only supports 2, 4, 8. 16, 32, 48, 64, 80, 96, 128, 224 for `n2d-standard-` machines: https://cloud.google.com/compute/docs/general-purpose-machines#n2d_machines)

Reply:
If the cpu count is invalid, please handle it in a most graceful way 
I'd print out warning and round it up or down to the nearest value
e.g., if cpuCount = 10, round it up to cpuCount = 16, which is the nearest valid value.
if cpuCount =  100000, round it down to 224 (max).

Reply:
I added some validation in the SubmitJob API:
func SubmitJob(ctx context.Context, c *app.RequestContext) {
	var req job.SubmitJobRequest
	var formReq FileParas
	err := c.BindAndValidate(&formReq)
	if err != nil {
		hlog.Errorf("[Job Handler]failed to parse parameters: %+v", err)
		utils.ReturnsJSONError(c, err)
		return
	}

	cpuCounts := []int64{2, 4, 8, 16, 32, 48, 64, 80, 96, 128, 224}

	inList := false
	closest := int64(0)
	minDistance := int64(math.MaxInt64)

	for _, n := range cpuCounts {
		if n == formReq.CPUCount {
			inList = true
			closest = n
			break
		}
		distance := int64(math.Abs(float64(formReq.CPUCount - n)))
		if distance < minDistance {
			minDistance = distance
			closest = n
		}
	}

	if !inList {
		formReq.CPUCount = closest
	}

Reply:
^ FYI, I debugged it but please double check if it works!